__tests__/building-highlights.test.js

```diff
@@ -1,12 +1,38 @@
 import React from "react";
 import BuildingHighlights from "../src/components/building-highlights/building-highlights.component";
 import renderer from "react-test-renderer";
+import { shallow } from "enzyme";
+import { BuildingId } from "../src/types/main";
+import { Buildings } from "../src/constants/buildings.data";
+import { CONCORDIA_RED, BUILDING_UNTAPPED } from "../src/constants/style";
 
-// Kind: Snapshot + Unit Tests
-// [US-2] Checks that all the highlights are rendering at their proper coordinates (latitude and longitude) also with proper styling
-describe("Building Highlight Snapshop", () => {
-  test("renders correctly", () => {
+describe("BuildingHighlights component", () => {
+  it("BuildingHighlights: snapshot renders correctly", () => {
     const tree = renderer.create(<BuildingHighlights />).toJSON();
     expect(tree).toMatchSnapshot();
   });
+
+  it("onBuildingTap: For Hall, before and after polygon press, check for polygon color and if function is called", () => {
+    const mockOnBuildingTap = jest.fn();
+    const wrapper = shallow(
+      <BuildingHighlights onBuildingTap={mockOnBuildingTap} />
+    );
+    const polygon = wrapper.find({ coordinates: Buildings[0].boundingBox });
+    expect(polygon.prop("fillColor", BUILDING_UNTAPPED));
+    polygon.simulate("press");
+    expect(mockOnBuildingTap).toHaveBeenCalledTimes(1);
+    expect(polygon.prop("fillColor", CONCORDIA_RED));
+    expect(mockOnBuildingTap.mock.calls[0][0]).toBe(BuildingId.H);
```

Comment:
Very nice, thank you for adding these checks for the actual value being passed to the mocked function :)

Reply:
ayy learned it from the best in the biz